### ISSUE TITLE
[Feature Request] Provide means to merge AcroForms of different documents

### ISSUE THREAD

**Issue by mara004** [pikepdf/pikepdf#207]
I'd like to preserve fillable forms if possible. For the case that only one source document has forms, I have this code so far:

```
# 'source' and 'dest' are pikepdf.Pdf objects
if hasattr(source.Root, 'AcroForm'):
    foreign_root = dest.copy_foreign(source.Root)
    if not hasattr(dest.Root, 'AcroForm'):
        dest.Root.AcroForm = foreign_root.AcroForm
```

What do I need to do if multiple documents do have forms? Is there a way to copy over all of them?
I tried `dest.Root.AcroForm = pikepdf.Dictionary(dict(dest.Root.AcroForm).update(dict(foreign_root.AcroForm)))`
but it doesn't work (breaks all forms).

**Comment by jbarlow83** [comment]
That'll be complicated. I think your best bet would be to use Acrobat to do the merge (I don't know if it can), unless you have a lot of files like this.

It requires a merge, not an update, and some nontrivial effort to identify and deduplicate form widget names. If there's Javascript involved it may be pretty much intractable.

**Comment by mara004** [comment]
Okay, I see. Thank you for the response. I already guessed it might be too complex.
It's not about specific documents, I just try to preserve content of the original document as far as possible in my code.

**Comment by mara004** [comment]
After looking at other programs, it turns out PDFsam Basic using sejda is capable of merging forms, just in case someone really needs it.

**Comment by jbarlow83** [comment]
Coming back to this one — merging AcroForms across documents is now supported directly.

The next release (**v10.9.0**) adds `Pdf.add_pages_from()`, which copies pages from another document and merges their form fields into the destination, handling the hard parts: carrying the `/AcroForm/Fields` tree, and auto‑renaming fields whose names collide between documents.

```python
from pikepdf import Pdf

dest = Pdf.new()
for path in ('a.pdf', 'b.pdf', 'c.pdf'):
    with Pdf.open(path) as src:
        result = dest.add_pages_from(src)   # works even when several docs have forms
        print(result.renamed_fields)        # fields renamed to resolve collisions
dest.save('merged.pdf')
```

It's page‑driven — only fields whose widgets are on the copied pages are carried over, so copying a subset won't drag in unrelated fields — and there's a `forms='strip'` mode if you want pages without any form data. (Your original `dest.Root.AcroForm = source.Root.AcroForm` attempt broke forms because it wholesale‑copies one document's field tree over another's; `add_pages_from` is the safe path.)

Merged to `main`; ships in v10.9.0. Thanks for the original request — reopen if your scenario isn't covered once it's out.
